libs/prelude/Prelude/Interfaces.idr

```diff
@@ -310,13 +310,13 @@ interface Foldable t where
   foldMap f = foldr ((<+>) . f) neutral
 
 ||| Combine each element of a structure into a monoid.
-%inline public export
+%inline %tcinline public export
```

Comment:
I can see tcinline making a difference on simple functions, does it actually
make a difference on these (potentially) recursive functions?

Reply:
Criteria was a function that is implemented through an interface function, so that if one uses such inlined function in an interface implementation, they won't get an unexpected termination checker error.

In particular case of `concat`, I've added and pushed an example of when code typechecks with added `%tcinline`s and doesn't without. It's arguable whether it is realistic to meet something like this in real code, but I think it's possible. Anyway, as far as I understand, it is safe in this particular case anyway.

Reply:
Thanks for the explanation!